### PR TITLE
Added configurable docker build directory for stress test

### PR DESCRIPTION
**Pull request by ckairen**

### CODE REVIEW

File: eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1 [hunk 1]
```diff
@@ -156,11 +150,24 @@ function DeployStressPackage(
     }
     $imageTag += "/$($pkg.Namespace)/$($pkg.ReleaseName):${deployId}"
 
-    $dockerFilePath = "$($pkg.Directory)/Dockerfile"
+    $dockerFilePath = if ($pkg.Dockerfile) {
+        [System.Io.Path]::Combine($pkg.Directory, $pkg.Dockerfile).ToString()
```
Comment (benbp) [inline eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1:154]: I would stick to using one method of joining paths, unless there's something I'm missing in the difference of usage here?

```suggestion
        Join-Path $pkg.Directory $pkg.Dockerfile
```

File: eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1 [hunk 2]
```diff
@@ -156,11 +150,24 @@ function DeployStressPackage(
...
     if ($pushImages -and (Test-Path $dockerFilePath)) {
         Write-Host "Building and pushing stress test docker image '$imageTag'"
         $dockerFile = Get-ChildItem $dockerFilePath
-        Run docker build -t $imageTag -f $dockerFile.FullName $dockerFile.DirectoryName
+        $dockerBuildFolder = if ($pkg.DockerBuildDir) {
+            Join-Path $pkg.Directory $pkg.DockerBuildDir
+        } else {
+            $dockerFile.DirectoryName.Trim()
+        }
+        $dockerBuildFolder = [System.IO.Path]::GetFullPath($dockerBuildFolder)
```
Comment (benbp) [inline eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1:168]: If you need to do `Trim()` here then it needs to be applied to both as either input could have a trailing slash.

```suggestion
        $dockerBuildFolder = if ($pkg.DockerBuildDir) {
            Join-Path $pkg.Directory $pkg.DockerBuildDir
        } else {
            $dockerFile.DirectoryName
        }
        $dockerBuildFolder = [System.IO.Path]::GetFullPath($dockerBuildFolder).Trim()
```

**Review by benbp — APPROVED** [review]

**Comment by richardpark-msft** [comment]
Do you have a test inside of here somewhere that makes sure this all works? It looks right to me....

**Comment by ckairen** [comment]
> Do you have a test inside of here somewhere that makes sure this all works? It looks right to me....

No, but I did use your PR's directory/file structure to test it locally.

**Pull request merged**